File: Arrays/00_Array_Definition/spec.rb
describe "new_array" do
  it "creates an array of numbers" do
    new_array(1,2,3,4).should eq([1,2,3,4])
  end
  it "creates an array of strings" do
    new_array("a", "b", "c", "d").should eq(["a", "b", "c", "d"])
  end
  it "creates an array of non-sequential objects" do
    new_array(1,4,2,3).should eq([1,4,2,3])
  end
end

describe "first_and_last" do
  it "creates new array with numbers" do
    first_and_last([1,2,3]).should eq([1,3])
  end
  it "creates new array with strings" do
    first_and_last(["a", "b", "c", "d"]).should eq(["a", "d"])
  end
end

File: 08-arrays/00_Array_Definition/spec.js
var expect    = require("chai").expect;
var challenge = require("./challenge.js");

describe("newArray", function() {
  it("creates an array of numbers", function() {
    expect(challenge.newArray(1,2,3,4)).to.eql([1,2,3,4]);
  });

  it("creates an array of strings", function() {
    expect(challenge.newArray("a", "b", "c", "d")).to.eql(["a", "b", "c", "d"]);
  });

  it("creates an array of non-sequential objects", function() {
    expect(challenge.newArray(1,4,2,3)).to.eql([1,4,2,3]);
  });
});

describe("firstAndLast", function() {
  it("creates new array with numbers", function() {
    expect(challenge.firstAndLast([1,2,3])).to.eql([1,3]);
  });

  it("creates new array with strings", function() {
    expect(challenge.firstAndLast(["a", "b", "c", "d"])).to.eql(["a", "d"]);
  });
});